src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -145,112 +145,31 @@ public BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorCon
-            .AppendLine();
-
-            scriptBuilder
-                .AddScriptToSetupSourceAndDestinationDirectories(
-                    sourceDir: sourceDir,
-                    temporaryDestinationDir: temporaryDestinationDir,
-                    destinationDir: destinationDir,
-                    hasUserSuppliedDestinationDir: hasUserSuppliedDestinationDir)
-                .AppendBenvCommand($"dotnet={context.ResolvedDotNetCoreRuntimeVersion}")
-                .AddScriptToRunPreBuildCommand(sourceDir: sourceDir, preBuildCommand: preBuildCommand)
-                .AppendLine("echo")
-                .AppendLine("dotnetCoreVersion=$(dotnet --version)")
-                .AppendLine("echo \"Using .NET Core SDK Version: $dotnetCoreVersion\"")
-                .AppendLine()
-                .AddScriptToRestorePackages(projectFile);
-
-            if (hasUserSuppliedDestinationDir)
-            {
-                scriptBuilder
-                    .AddScriptToPublishOutput(
-                        projectFile: projectFile,
-                        buildConfiguration: buildConfiguration,
-                        finalDestinationDir: destinationDir)
-                    .AddScriptToRunPostBuildCommand(
-                        sourceDir: sourceDir,
-                        postBuildCommand: postBuildCommand);
-            }
-            else
+            var templateProperties = new DotNetCoreBashBuildSnippetProperties
             {
-                scriptBuilder
-                    .AddScriptToBuildProject(projectFile)
-                    .AddScriptToRunPostBuildCommand(
-                        sourceDir: sourceDir,
-                        postBuildCommand: postBuildCommand);
-            }
-
-            SetStartupFileNameInfoInManifestFile(context, projectFile, manifestFileProperties);
+                ProjectFile = projectFile,
+                Configuration = GetBuildConfiguration(),
+            };
 
-            scriptBuilder
-                .AddScriptToCreateManifestFile(
-                    manifestFileProperties,
-                    manifestDir: _cliOptions.ManifestDir,
-                    finalDestinationDir: destinationDir)
-                .AppendLine("echo Done.");
+            var script = TemplateHelper.Render(
+                TemplateHelper.TemplateResource.DotNetCoreSnippet,
+                templateProperties,
+                _logger);
 
             return new BuildScriptSnippet
             {
-                BashBuildScriptSnippet = scriptBuilder.ToString(),
-                IsFullScript = true,
+                BashBuildScriptSnippet = script,
+                BuildProperties = manifestFileProperties,
+                PlatformInstallationScriptSnippet = installationScriptSnippet,
+
+                // Setting this to false to avoid copying files like '.cs' to the destination
+                CopySourceDirectoryContentToDestinationDirectory = false,
```

Comment:
Just a little confused why we're setting it to be false while within `BuildScriptSnippet.cs` we let it default be true

Reply:
> Just a little confused why we're setting it to be false while within BuildScriptSnippet.cs we let it default be true

because default is true, means for all platforms we are copying the source directory content, except dotnetcore apps and multiplatform apps where we do dotnet publish/build

Reply:
In case of Node and Python platforms we copy the source files to the target folder and they get interpreted & compiled during run of an app where as in case of .NET Core we compile from source files to get the output and run using them so we do not need the source files (ex: *.cs).